### ISSUE TITLE
Unencrypted MMS Messages Received and Displayed Twice After Delay

### ISSUE THREAD

**Issue by shakurass** [WhisperSystems/Signal-Android#6112]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
I'm experiencing an issue with regular unencrypted MMS messages. Signal will receive and display the MMS message normally however after another 3 days it will display the same message a second time even though the sender only sent it along once. I switched my phone back to the stock SMS app (Google Messenger) for a while to see if I could reproduce the same behavior. Google Messenger displays MMS message a single time as expected.

This behavior has been consistent for me since I installed Signal and it does not seem to matter what phone OS (Android or iOS) or messaging application (iMessage, etc.) the sender is using.

Possibly related issues:
https://github.com/WhisperSystems/Signal-Android/issues/6003
https://github.com/WhisperSystems/Signal-Android/issues/5525

### Steps to reproduce
- Get a friend to send along a single MMS message
- Notice that the MMS message is received and displayed by your phone normally
- Wait 3 days
- Notice that the MMS message is received and displayed by Signal a second time

**Actual result:** Single MMS messages are being received and displayed twice by Signal.
**Expected result:** MMS messages are only received and display once in Signal.

### Screenshots
https://goo.gl/photos/jFyRb3jeS8s9sshZ8

Notice the sent and received dates of both images.

### Device info
**Device:** LGE Nexus 5X
**Android version:** 7.1.1
**Signal version:** 3.27.1

### Link to debug log
https://gist.github.com/anonymous/d166776852cf019e7c9552e2ee2ebfb2

Log was captured after MMS message was received and displayed the 2nd time.

**Comment by brastewa** [comment]
I have had the app for over a year and I have had the same problem as listed above. This has been a problem for me since day 1.

**Device info**

**Device:** Samsung Galaxy Note 4
**Android version:** 6.0.1
**Signal version:** 3.28.4

**Comment by Genis-Sage** [comment]
I also have this problem for a very long time.
(so also with older versions of Signal and also on my old phone)

However, I don't receive MMS two times but four times.
The time between each MMS is about a day.
After the forth time, I receive a SMS from my provider,
that there is a MMS on its server,
and that I can get it on a website, entering my phone number and a code sent in the SMS.
(This works.)

Some time ago, I also switched back to Google Messenger, after I got an MMS in signal.
After I received the same MMS the next day with Google Messenger,
there were no more repetitions.
(Then I could go back to Signal as SMS/MMS app.)

For me, without a coding background, it some how looks,
as Signal misses some kind of answer back to the MMS server,
stating that the download was completed.

**Device info**

**Device:** Blackberry Priv 
**Android version:** 6.0.1
**Signal version:** 4.0.0
**Provider:** T-mobile Germany / Deutsche Telekom
